boards/arm/v2m_musca/v2m_musca.dts

```diff
@@ -0,0 +1,36 @@
+/dts-v1/;
+
+#include <arm/armv8-m.dtsi>
+
+#if defined (CONFIG_ARM_NONSECURE_FIRMWARE)
+#include "v2m_musca-nonsecure.dtsi"
+#else
+#include "v2m_musca-secure.dtsi"
+#endif
+
+/ {
+	compatible = "arm,v2m-musca";
+	#address-cells = <1>;
+	#size-cells = <1>;
+
+	chosen {
+		zephyr,console = &uart1;
+		zephyr,sram = &sram0;
```

Comment:
This one will work, if sram0 is defined differently in the two .dtsi files. If not, we need to implement some kind of partition, here (in the Board definition) and choose the right sram for the secure and the non-secure image.